apps/aecore/src/aec_parent_chain_cache.erl

```diff
@@ -312,59 +314,31 @@ state_to_map(#state{child_start_height = StartHeight,
        blocks             => Blocks,
        top_height         => TopHeight}.
 
-target_parent_height(#state{child_start_height = StartHeight,
-                            child_top_height   = ChildHeight}) ->
-    ChildHeight + StartHeight.
+target_parent_heights(#state{parent_target_fun  = ParentTargetFun,
+                            child_top_height    = ChildHeight}) ->
+    ParentTargetFun(ChildHeight).
 
-min_cachable_parent_height(#state{max_size   = CacheSize,
-                                  top_height = ParentTop} = State) ->
-    min(target_parent_height(State), ParentTop - CacheSize ).
+min_cachable_parent_height(State) ->
+    [MinHeight|_] = lists:sort(target_parent_heights(State)),
+    MinHeight.
 
-post_block(Block, #state{max_size = MaxSize,
-                         top_height = TopHeight} = State0) ->
-    TargetHeight = target_parent_height(State0),
+max_cachable_parent_height(State) ->
+    [MaxHeight|_] = lists:reverse(lists:sort(target_parent_heights(State))),
+    MaxHeight.
+
+post_block(Block, #state{top_height = TopHeight} = State0) ->
```

Comment:
I am unsure about the working of the cache, but it seems that the privious version did consider a number of weird cases, potentiallly attacks or things spotted. These cases are now all ignored. Probably right but hard to see consequences of this, because tests for cache are disabled in this commit.

Reply:
Only the tests related to commitments should be disabled. Which tests should I add back?